client/src/client.rs

```diff
@@ -166,7 +202,82 @@ impl<DB: Database> Client<DB> {
 
         let res = self.node.write().await.sync().await;
         if let Err(err) = res {
-            warn!("consensus error: {}", err);
+            warn!(
+                "failed to sync consensus node with checkpoint: {:?}. Error: {}",
+                self.node.read().await.config.checkpoint,
+                err
+            );
+            let fallback = if let Some(fallback) = &self.fallback {
```

Comment:
I think a better order of operations to do here is that if the initial sync fails:

1. if user has set a fallback url and has set the load from fallback flag, then use just that fallback url to fetch a checkpoint. If it fails, do not continue.
2. If user has not set a fallback url and has set the load from fallback flag, then use all the ones from repo to try.
3. If load from fallback is not set, always fail.

The reason I think this is useful, is that someone might want to set their fallback url in their config file, but never have that get used unless they have problems syncing. If they have a problem, they can use their prefered url to sync by just adding `-l`.

I also thing if someone explicitly sets a fallback url, they are sort of saying "I trust this one but not the others", so we shouldn't continue on to use the others.

Basically I think `-l` should act as a toggle and `-f` should act as a url override.

Reply:
Actually the more I think about it I think the original way was better. What do you think?